### PR TITLE
Test cleanup

### PR DESCRIPTION
**Pull request by holgerd77**
This half-cleaned [tests](https://github.com/ethereumjs/ethereumjs-vm/tree/master/tests) directory together with the outdated `runAll()` test run function in `tester.js` is for quite some time some reoccurring source of confusion, latest during the work on #406.

This PR does some cleanup here:

### Separate `manual` directory with the manual `constantionpleSstoreTest` test file

This was once written by Vinay since the official test suite was not yet available. Tests still pass so would be a bit a pity to throw away. For the current situation this extra `manual` command might be a bit too much. Reasoning would be nevertheless that there is then a place for similar tests in the future if necessary and things have some place then and there will be some higher chance that structure won't get bloated on stuff like this again.

Update: Tests deleted after discussion.

### Included `hooked.js` into the API test suite

These tests are still working. We can remove/adopt depending on what we do with this part of the API. For now the code is still there so this fits better in the API directory than having some (too prominent) separate place.

### Integrated genesis hash test into the API test suite

This test was also floating around, not used for quite some time but still working. Have adopted and added to the API test suite. Genesis functionality of `StateManager` is also not sure to be kept on final extracted version. Nevertheless for the moment it's still there and used.

### Replaced outdated runAll() test run code with info message

When `npm run test` is run this is now showing a Command-not-used message. The code within the function `runAll()` is commented out. This was/is some major source of confusion for newcomers who tend to integrate their new tests there.

Update: Function deleted after discussion.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "testBlockchain": "npm run build:dist && node --stack-size=1500 ./tests/tester -b --fork='Constantinople' --dist --excludeDir='GeneralStateTests'",
     "testBlockchainGeneralStateTests": "npm run build:dist && node --stack-size=1500 ./tests/tester -b --dist --dir='GeneralStateTests'",
     "testAPI": "tape ./tests/api/*.js",
+    "testManual": "tape ./tests/manual/*.js",
```
Comment (s1na) [inline package.json]: I assume the standard tests cover sstore exhaustively; if that's the case, couldn't we drop `constantinopleSstoreTest.js` (and `manual` for now)? I know deleting code is hard, but I feel like it has served its purpose :)

File: tests/tester.js [hunk 2]
```diff
@@ -246,12 +245,14 @@ function runTests (name, runnerArgs, cb) {
 }
 
 function runAll () {
-  require('./tester.js')
+  console.log('[INFO] Generic test cmd not used. See package.json for more specific test run cmds.')
```
Comment (s1na) [inline tests/tester.js]: Same here, I think it should be fine to just delete this, as the tests don't need to maintain backwards compatibility or anything. In `package.json`, the `test` script is calling this here, not sure what it should be changed to, maybe `testAPI`?

Comment (holgerd77) [reply tests/tester.js]: Wasn't sure in both cases, but I am feeling a bit thankful for both comments now. 😂 Will delete both.

File: package.json (hunk 1 above)
Comment (holgerd77) [reply package.json]: Have deleted and pushed.

File: tests/tester.js (hunk 2 above)
Comment (holgerd77) [reply tests/tester.js]: Have also deleted. I have now placed the info message directly into the npm test command, think that's most practically useful especially for newcomers to have some guidance when they start with `npm run test`. Wasn't any good reasoning (as well as use case) to include some specific test run command in here.

**Comment by holgerd77** [comment]
Just x-checked, the decrease in test coverage is from the `lib/hooked.js` file now included in the line count, so with this PR relevant lines covered increases but also the total line count (see [PR build](https://coveralls.io/builds/21566636/) vs [master build](https://coveralls.io/builds/21506762).

So no actual coverage decrease and we can merge if everything else is ok.

**Comment by holgerd77** [comment]
Already reviewed by @s1na, no production code touched, will merge.

**Pull request merged**